### DISCUSSION TITLE
Problem with bilingual video

### DISCUSSION THREAD

**Discussion by 1001ruchka** [Purfview/whisper-standalone-win#200] [Q&A] [answered]
Video in 2 languages - English (mostly) and Russian.
In the Q&A section participants ask questions in Russian, the speaker answers in English. Sometimes if there is an English piece of speech after the Russian speech (Whisper tries to translate the Russian speech into English), it may not be recognized.
Example:
Timing 03:22:37,745 to 03:23:02,660

[whisper log.zip](https://github.com/Purfview/whisper-standalone-win/files/14422824/whisper.log.zip)

YouTube subtitles of the same video.
[[English (auto-generated)] ImTT Advanced Workshop Day 2 eng cut [DownSub.com].srt.zip](https://github.com/Purfview/whisper-standalone-win/files/14422870/English.auto-generated.ImTT.Advanced.Workshop.Day.2.eng.cut.DownSub.com.srt.zip)

command line: whisper-faster.exe --language en --model "large-v2" --compute_type=float16 --sentence --beep_off -f=all -prompt auto "video.mp4"

Whisper-Faster r172.4 windows
There were also errors on 182.1, but did the logs on 172.4

**Comment by Purfview** [comment, marked as answer]
Whisper doesn't support transcription of multilingual audio.

**Reply by 1001ruchka** [reply]
But he's doing it now. It's just that there are mistakes with missing fragments of English speech coming right after Russian speech. But they are about 5-7 minutes per 3.5 hours of video.

**Reply by 1001ruchka** [reply]
Please note the task is not to make subtitles in 2 languages. But to make subtitles in English, leaving out Russian speech. 
Just like YouTube generated English subtitles for this video.

**Reply by 1001ruchka** [reply]
Whisper, on the other hand, tries to translate Russian speech into English, although the translation mode is not enabled. And it skips some of the fragments in English following the Russian speech.

**Reply by Purfview** [reply]
> But he's doing it now.

As you can see, doing with the issues, because it doesn't support multilingual audio.

Try `-condition=false`, `-prompt=None`.

**Reply by 1001ruchka** [reply]
Unfortunately it didn't help.

[whisper log 2.zip](https://github.com/Purfview/whisper-standalone-win/files/14424967/whisper.log.2.zip)

**Reply by 1001ruchka** [reply]
Found an interesting tip on huggingface and it seems to have worked. I will have to compare the quality of the result carefully.

> Not exactly a solution but it worked best:
> 
> set language to auto and then put translate as task.

whisper-faster.exe --model "large-v2" --compute_type=float16 --sentence -prompt=auto --task translate --beep_off -f=all "video.mp4"

[whisper log 3.zip](https://github.com/Purfview/whisper-standalone-win/files/14434127/whisper.log.3.zip)

Timing 03:22:37,745 to 03:23:02,660 - Speech recognized.

**Reply by Purfview** [reply]
With what version "log3" was done?

Btw, your archives are done on macOS, but there are no "r172.4" for mac. How do you run it?

**Reply by 1001ruchka** [reply]
Version 182.1.
I'm using a cloud-based Windows Server 2022 with Nvidia Tesla A10 or T4. I'm connecting via RDP from my Macbook. The files are copied and archived on the Mac.

**Reply by Purfview** [reply]
Check with  `--language en -prompt None`

**Reply by 1001ruchka** [reply]
On first impression - it worked!

[whisper log 4.zip](https://github.com/Purfview/whisper-standalone-win/files/14460390/whisper.log.4.zip)

Command line: `whisper-faster.exe --language en --model "large-v2" --compute_type=float16 --sentence -prompt None  --beep_off -f=all "video.mp4"`

**Reply by Purfview** [reply]
Then in "log 3" it worked not because of translate task.
There "auto" was switched to "None" because language wasn't set.

**Reply by 1001ruchka** [reply]
New test. Unfortunately the command line argument `-promt None` did not help, but `--task translate` worked.
`-promt None`
[Promt None log.zip](https://github.com/Purfview/whisper-standalone-win/files/14484421/Promt.None.log.zip)
`--task translate`
[Whisper-Faster translate.zip](https://github.com/Purfview/whisper-standalone-win/files/14484425/Whisper-Faster.translate.zip)
Time: 00:03:55,460  to 00:04:20,960

Command lines: 
`whisper-faster.exe --language en --model "large-v2" --compute_type=float32 --sentence -prompt None --beep_off -f=all --print_progress "video2.mp4"`

`whisper-faster.exe --language en --model "large-v2" --compute_type=float32 --sentence -prompt None --task translate --beep_off -f=all --print_progress "video2.mp4"`

Float32 is used because the test was tested on a computer with GTX 970 video card.

**Reply by Purfview** [reply]
You just observing random behaviour.
Once again: "Whisper doesn't support transcription of multilingual audio."

**Reply by 1001ruchka** [reply]
I have used whisper quite successfully with bilingual videos.
Of the 90 minutes of bilingual audio - he missed about 30 seconds.  The above test is an fragment of the full video.
With the -`-task translate` parameter there are no skips.
